### ISSUE TITLE
Adding a utility to penalize majority class pixels in the Segmentation tutorial

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/tensorflow#48714]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

## URL(s) with the issue:

Please provide a link to the documentation entry, for example:
https://www.tensorflow.org/tutorials/images/segmentation

## Description of issue (what needs changing):

It's not really an issue, a suggestion rather. 

### Clear description

Semantic segmentation datasets can be highly imbalanced meaning that particular class pixels can be present more inside images than that of other classes. Since segmentation problems can be treated as per-pixel classification problems we can deal with the imbalance problem by weighing the loss function to account for this. It's a simple and elegant way to deal with this problem. Other solutions include always ensuring that a batch of samples (during training) always contain some proportion (which is prefixed) of positive classes. 

However, TensorFlow does not yet support the `class_weight` argument in `model.fit()` for targets that are 3D (for segmentation problems, we are essentially predicting a map of shape `[batch_size, height, width, nb_channels]`). One way to get around this problem is to use `sample_weight` instead. But then again, it's not very clear as to how to do that properly particularly with `tf.data` pipelines. 

Multiple folks have tried several hacks to get around this problem but it keeps coming back (see [here](https://github.com/keras-team/keras/issues/3653)). Therefore, I think the tutorial under question is a perfect opportunity to demonstrate the use case. 

Cc: @MarkDaoust

**Comment by deeb02** [comment]
That sounds like a reasonable idea at face value. Why not submit a PR with that change?

**Comment by sayakpaul** [comment]
Since I don't have the workaround yet, I wanted to pass this on to the team responsible for authoring the tutorial. If that was not the case I would have submitted the PR.

**Comment by deeb02** [comment]
I closed it as a Keras issue, but it may make sense as a documentation issue.

**Comment by MarkDaoust** [comment]
I'll take  look. It seems easy to implement a workaround.

**Comment by sayakpaul** [comment]
Looking forward to it. FWIW, I don't think it's a docs issue. In fact, the tutorial is very clear and well written. It's rather an incremental suggestion to accomplish a use case.

**Comment by bhack** [comment]
@MarkDaoust I think the thread is alive since 2016 at https://github.com/keras-team/keras/issues/3653

**Comment by sayakpaul** [comment]
^ which is what I mentioned in my initial post as well.

**Comment by MarkDaoust** [comment]
I wonder why the "Fix:" comment on the commit didn't close the bug.

**Comment by bhack** [comment]
@MarkDaoust  Cross repositories is not in the format: https://github.blog/2013-03-18-closing-issues-across-repositories/?

**Comment by bhack** [comment]
As I don't see your direct commits in this repo other than an old commit in 2015 but with another email.  Do you have push right here?

**Comment by bhack** [comment]
Ok I've checked with your current email and all seems fine. Very strange.

**Comment by MarkDaoust** [comment]
Yeah, I think you're right that it's just a permissions issue.

**Comment by bhack** [comment]
Yes probably is ` MarkDaoust authored and Copybara-Service`

**Comment by bhack** [comment]
We don't have that commiter bot here `git log --committer="Copybara-Service"`. it is used in the Docs repository. This is why Github don't close your cross ticket as the permission check it is related to the committer not to the author (you).

**Comment by sayakpaul** [comment]
@MarkDaoust thank you for your contribution for the section:

https://www.tensorflow.org/tutorials/images/segmentation#optional_imbalanced_classes_and_class_weights

We now at least have a section in the official docs providing a way to get around this important problem.
